src/commands/registries/pullImages.ts

```diff
@@ -6,33 +6,32 @@
 import { Terminal } from "vscode";
 import { IActionContext } from "vscode-azureextensionui";
 import { ext } from '../../extensionVariables';
-import { DockerHubNamespaceTreeItem } from '../../tree/registries/dockerHub/DockerHubNamespaceTreeItem';
+import { registryExpectedContextValues } from "../../tree/registries/registryContextValues";
 import { RegistryTreeItemBase } from '../../tree/registries/RegistryTreeItemBase';
 import { RemoteRepositoryTreeItemBase } from "../../tree/registries/RemoteRepositoryTreeItemBase";
-import { RemoteTagTreeItemBase } from '../../tree/registries/RemoteTagTreeItemBase';
+import { RemoteTagTreeItem } from '../../tree/registries/RemoteTagTreeItem';
 import { logInToDockerCli } from "./logInToDockerCli";
 
 export async function pullRepository(context: IActionContext, node?: RemoteRepositoryTreeItemBase): Promise<void> {
     if (!node) {
-        node = await ext.registriesTree.showTreeItemPicker<RemoteRepositoryTreeItemBase>(RemoteRepositoryTreeItemBase.allContextRegExp, context);
+        node = await ext.registriesTree.showTreeItemPicker<RemoteRepositoryTreeItemBase>(registryExpectedContextValues.all.repository, context);
     }
 
     await pullImages(context, node.parent, node.repoName + ' -a');
 }
 
-export async function pullImage(context: IActionContext, node?: RemoteTagTreeItemBase): Promise<void> {
+export async function pullImage(context: IActionContext, node?: RemoteTagTreeItem): Promise<void> {
     if (!node) {
-        node = await ext.registriesTree.showTreeItemPicker<RemoteTagTreeItemBase>(RemoteTagTreeItemBase.allContextRegExp, context);
+        node = await ext.registriesTree.showTreeItemPicker<RemoteTagTreeItem>(registryExpectedContextValues.all.tag, context);
     }
 
-    await pullImages(context, node.parent.parent, node.fullTag);
+    await pullImages(context, node.parent.parent, node.repoNameAndTag);
 }
 
 async function pullImages(context: IActionContext, node: RegistryTreeItemBase, imageRequest: string): Promise<void> {
```

Comment:
Also the file name is confusing.

Reply:
Y'all there's only so much refactoring/file-renaming a person can do at a time 😂

With the caveat that I didn't name this, I think it's named like this because pulling a repository is technically just pulling multiple images at once.